### ISSUE TITLE
`oclif readme` produces unstable content

### ISSUE THREAD

**Issue by moltar** [oclif/oclif#1223]
Do you want to request a *feature* or report a *bug*?
-----------------------------------------------------

Bug

What is the current behavior?
-----------------------------

`oclif readme` renders CLI help as-is.

If a command has a default value on a flag that is dynamic (e.g. `process.cwd()`), the value will be rendered into markdown.

What is the expected behavior?
------------------------------

Not sure.

**Comment by mdonnalley** [comment]
@moltar Can you better explain the issue? Is it that the value of `process.cwd()` is put into the readme? If you want to avoid that you should use `defaultHelp` instead of `default`. 

`defaultHelp` is specifically designed for flags that contain a dynamic default value, which shouldn't be put into the generated readme or the oclif.manifest.json

Example: https://github.com/salesforcecli/sf-plugins-core/blob/5.0.13/src/flags/orgFlags.ts#L181

I realize that this isn't documented anywhere - I've made a note to update it

**Comment by moltar** [comment]
> Is it that the value of process.cwd() is put into the readme?

Yes, that's right. The current value of the cwd gets added to the readme.

> If you want to avoid that you should use defaultHelp instead of default.

Hm, I think I tried that too. But let me try again.

**Comment by moltar** [comment]
Can't seem to get it to work.

Code:

```ts
  static flags = {
    dir: Flags.string({
      char: 'd',
      description: 'Location where to create the project directory.',
      default: async () => process.cwd(),
      defaultHelp: async () => '$PWD',
    }),
  }
```

Result:

```md
USAGE
  $ slp new NAME [-d <value>]

ARGUMENTS
  NAME  Project name. A directory with this name will be created.

FLAGS
  -d, --dir=<value>  [default: /Users/foo/bar/packages/foo-cli] Location where to create
                     the project directory.

DESCRIPTION
  Bootstraps and initializes a new project.
```

**Comment by mdonnalley** [comment]
That flag configuration works for me on the [hello-world template](https://github.com/oclif/hello-world). It's possible that you just need to compile before running `oclif readme` since that commands only looks at the compiled code

**Comment by moltar** [comment]
> looks at the compiled code

Indeed that was the issue!!

Thank you! 😁
